openfpga/src/utils/report_reference.cpp

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************************
+ * Output internal structure of module graph to XML format
+ ***************************************************************************************/
+/* Headers from system goes first */
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <string>
+
+/* Headers from vtrutil library */
+#include "vtr_assert.h"
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+/* Headers from openfpgautil library */
+#include "command_exit_codes.h"
+#include "openfpga_digest.h"
+#include "report_reference.h"
+
+/* begin namespace openfpga */
+namespace openfpga {
+
+/********************************************************************
+ * Top-level function
+ *******************************************************************/
+int report_reference(const char* fname, const std::string& module_name,
+                     const ModuleManager& module_manager,
+                     const bool& include_time_stamp, const bool& verbose) {
+  vtr::ScopedStartFinishTimer timer("Report reference");
+
+  std::fstream fp;
```

Comment:
Suggest to split the functions into two subfunctions, one is for reporting to a file, the other is for reporting to stdout (VTR_LOG)